spec/models/form_spec.rb

```diff
@@ -32,6 +32,48 @@
     end
   end
 
+  describe "next_incomplete_section_redirect_path" do
+    let(:case_log) { FactoryBot.build(:case_log, :in_progress) }
+    let(:subsection) { form.get_subsection("household_characteristics") }
+    let(:later_subsection) { form.get_subsection("local_authority") }
+
+    context "when a user is on the check answers page for a subsection" do
+      before do
+        case_log.tenant_code = "123"
+        case_log.age1 = 35
+        case_log.sex1 = "Male"
+        case_log.other_hhmemb = 0
+      end
+
+      it "returns the first page of the next incomplete subsection" do
+        expect(form.next_incomplete_section_redirect_path(subsection, case_log)).to eq("armed-forces")
+      end
+
+      it "returns the first page of the next incomplete subsection (skipping completed subsections)" do
+        case_log.armedforces = "No"
+        case_log.illness = "No"
+        case_log.housingneeds_a = "Yes"
+        case_log.la = "York"
+        case_log.illness_type_1 = "Yes"
+        expect(form.next_incomplete_section_redirect_path(subsection, case_log)).to eq("tenancy-code")
+      end
+
+      it "returns the next incomplete section by cycling back around if next subsections are completed" do
```

Comment:
Am I understanding this right that if I have:

Section A - Incomplete
Section B - Complete
Section C - Incomplete

I'm on Check Answers page for Section B - the next incomplete section link should take me to C not A?

Reply:
No, I couldn't think of a nice way to phrase this but if you had

Section A - Incomplete
Section B - Incomplete
Section C - Complete
Declaration Section - Incomplete (requires all other sections completed)

And you're on Check Answers page for Section C instead of going to the declaration you should loop back around to section A

Reply:
The test before this one is checking for the behaviour you described

Reply:
Cool, that makes sense - was just checking that the behaviour I described _is_ the intended behaviour.